src/main/java/com/serotonin/bacnet4j/type/Encodable.java

```diff
@@ -448,6 +443,9 @@ protected static Encodable readANY(final ByteQueue queue, final ObjectType objec
         }
 
         if (propertyArrayIndex == null && def.isCollection()) {
+            if (def.isArray()) {
+                return readArray(queue, def.getClazz(), contextId);
+            }
```

Comment:
This is the main change. It says that if the property is defined (spec) as an array then create a `BACnetArray` (a subclass of SequenceOf)